src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs

```diff
@@ -370,9 +501,9 @@ public void AssignTemplatesNamesAndCompile(MethodCompiler compiler, PEModuleBuil
                 }
             }
 
-            var synthesizedDelegates = ArrayBuilder<SynthesizedDelegateSymbol>.GetInstance();
+            var synthesizedDelegates = ArrayBuilder<AnonymousDelegateTemplateSymbol>.GetInstance();
             GetCreatedSynthesizedDelegates(synthesizedDelegates);
-            if (synthesizedDelegates.Count > 0)
+            if (delegates.Count > 0 || synthesizedDelegates.Count > 0)
```

Comment:
>delegates [](http://example.com/codeflow?start=16&length=9)

Does the body of the ```if```supposed to do something about these delegates? Like visit them, etc.?

Reply:
Added a call to `compiler.Visit()` for completeness, although it's not clear whether it is needed for synthesized delegates.

Reply:
> Added a call to ```compiler.Visit()``` for completeness, although it's not clear whether it is needed for synthesized delegates.

We visit non-delegates too. I'd say it is better to visit everything.